drivers/serial/uart_nrfx_uarte2.c

```diff
@@ -385,6 +384,15 @@ static int api_tx(const struct device *dev, const uint8_t *buf, size_t len, int3
 	nrfx_err_t err;
 	bool hwfc;
 
+#if CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	(void)pm_device_state_get(dev, &state);
+	if (state != PM_DEVICE_STATE_ACTIVE) {
+		return 0;
```

Comment:
Shouldn't an error be returned here? Since API says that: https://github.com/zephyrproject-rtos/zephyr/blob/28f5d24b3eb08db8447d5ef2f76f8f0433153d97/include/zephyr/drivers/uart.h#L1268-L1269 one could expect that callback being called since this function was successful.

Reply:
Added -ECANCELED error. Though, legacy shim was also returning (but there timeout was started in case of non zero timeout so there was abort event then but nothing when no timeout was provided)

Reply:
> Added -ECANCELED error.

Did you forget to push the branch?

Reply:
yes. Pushed now.